data/scripts/dependency-resolver.py

```diff
@@ -80,10 +81,23 @@ def find_makefile_var(self, v):
             return var["value"]
         return ""
 
-def run_command(cmd):
+    def add_log_content(self, line):
+        self.log_content.append(line)
+
+    def write_log_content(self, path):
+        if not self.log_content:
+            return
+
+        with open(path, "w+") as f:
+            for line in self.log_content:
+                f.write("%s\n" % line)
+
+def run_command(cmd, context):
     try:
         output = subprocess.check_output(cmd, stderr=subprocess.STDOUT,
                                          shell=True, universal_newlines=True)
+        context.add_log_content("Running command: %s" % cmd)
+        context.add_log_content("Command output:\n%s" % (output if output else "None"))
```

Comment:
not enough.

as per above, write directly to the file. And write the command to be executed BEFORE executing it. Also write the `os.environ` sorted by key as it impacts how the execution happens.

It would be nice to get more information about the errors... maybe `subprocess.check_output()` is not enough, maybe it is, just need to log from the exception. But the exit code and error messages would be nice to have, if possible separate stdout from stderr, prefixing lines.